File: QuanLyPhongKham.Server/src/QuanLyPhongKham.DTOs/Responses/ChiTietDonThuocResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuanLyPhongKham.DTOs.Responses
{
    public class ChiTietDonThuocResponse
    {
        public Guid MaCTDT { get; set; }
        public Guid? MaThuoc { get; set; }
        public string? TenThuoc { get; set; }
        public decimal DonGia { get; set; }
        public int SoLuong { get; set; }
        public string? LieuDung { get; set; }
        public decimal ThanhTien => DonGia * SoLuong;
    }
}

File: QuanLyPhongKham.Client/src/interfaces/responses/ChiTietDonThuocResponse.js
/**
 * @typedef {Object} ChiTietDonThuocResponse
 * @property {string} maCTDT
 * @property {string} maThuoc
 * @property {string} tenThuoc
 * @property {number} donGia
 * @property {number} soLuong
 * @property {string} lieuDung
 */

export const createChiTietDonThuocResponse = (
  maCTDT = "",
  maThuoc = "",
  tenThuoc = "",
  donGia = 0,
  soLuong = 0,
  lieuDung = "",
) => ({
  maCTDT,
  maThuoc,
  tenThuoc,
  donGia,
  soLuong,
  lieuDung,
});
